### PR TITLE
Simplify logic of REST request suffix parsing.

### PR DESCRIPTION
**Pull request by domob1812**
This patch changes the way the suffix (giving the requested data format) is parsed for REST requests.  Before, the string was split at '.' characters and it was assumed that the second part was the suffix.

Now, we look for the last dot and use that to determine the suffix.  This allows for strings that contain dots (not used now, though), and seems, in general, to be clearer and more intuitive.

### CODE REVIEW

**Comment by jgarzik** [comment]
ut ACK - I like it

**Comment by sipa** [comment]
utACK

**Comment by dcousens** [comment]
utACK

**Comment by jonasschnelli** [comment]
Definitively cleaner.
utACK.

**Comment by laanwj** [comment]
Looks good to me.
utACK (but the tests cover this).

File: src/rest.cpp [hunk 1]
```diff
@@ -71,15 +71,24 @@ static bool RESTERR(HTTPRequest* req, enum HTTPStatusCode status, string message
...
-                return rf_names[i].rf;
+    const std::string::size_type pos = strReq.rfind('.');
+    if (pos == std::string::npos)
+    {
+        param = strReq;
+        return rf_names[0].rf;
     }
 
+    param = strReq.substr(0, pos);
+    const std::string suff(strReq, pos + 1);
```
Comment (laanwj) [inline src/rest.cpp]: Didn't even know std::string had a substring constructor. Nice.

**Pull request merged**